src/ami-set.js

```diff
@@ -51,6 +55,49 @@ AmiSet.listAmiSets = async function () {
 };
 
 /**
+ * Checks if the AMIs from the amiSet are valid or not.
+ */
+
+async function checkAmi(ctx, region, ami, vtype) {
+  let request;
+  let missing = [];
+  let virtualizationType = '';
+
+  if (ami) {
+    request = {
+      ImageIds: [ami],
+    };
+    let exists = await amiExists(ctx.ec2[region], ami);
+    if (exists) {
+      virtualizationType = await getVirtualizationType(ctx.ec2[region], ami);
+      if (virtualizationType !== vtype) {
+        missing.push({imageId: ami, region: region, virtualizationType: virtualizationType});
+      }
+    } else {
+      missing.push({imageId: ami, region: region, virtualizationType: virtualizationType});
+    };
+  }
+  return missing;
+};
+
+AmiSet.validate = async function (ctx, amiSet) {
+  let missing = [];
+  let exists = false;
+
+  await Promise.all(amiSet.amis.map(async (def) => {
+
+    if (def.hvm) {
+      missing = missing.concat(await checkAmi(ctx, def.region, def.hvm, 'hvm'));
+    }
+    if (def.pv) {
+      missing = missing.concat(await checkAmi(ctx, def.region, def.pv, 'paravirtual'));
+    }
+  }));
+
+  return missing;
```

Comment:
Maybe return an object like this:
```javascript
return {
  valid: valid,
  invalidAmis: missing
};

```
To track the valid, what you could do is something like `let valid = true` and then only set it to false when you have a condition that invalidates validity.

As well, instead of returning the list of missing ones, let's return a more general set of information.  Missing implies that the AMI is invalid, but what if it fails because of the wrong virtualization type instead of abesnce?  For the invalidAmis, a list of objects like `{region: ..., ami: ..., vtype: ...}` would be really helpful.